src/libraries/System.Text.Json/gen/JsonSourceGenerator.Parser.cs

```diff
@@ -1026,7 +1026,11 @@ private TypeGenerationSpec GetOrAddTypeGenerationSpec(Type type, JsonSourceGener
 
                         bool propertyOrderSpecified = false;
 
-                        for (Type? currentType = type; currentType != null; currentType = currentType.BaseType)
+                        Type[] sortedTypeHierarchy = type.IsInterface
+                            ? type.GetSortedInterfaceHierarchy()
+                            : type.GetSortedClassHierarchy();
```

Comment:
would make sense moving these lines into a helper inline method? I am seeing it is repeated somewhere else.